modules/elements-color-picker/assets/js/editor/commands/show-swatches.js

```diff
@@ -219,53 +290,31 @@ export class ShowSwatches extends CommandBase {
 					picker.style.setProperty( '--right', '0' );
 				}
 
-				picker.classList.remove( 'e-picker-hidden' );
+				picker.classList.remove( this.classes.hidden );
 			},
 			root: this.container.view.$el[ 0 ],
 			offset: `0px -${ parseInt( picker.getBoundingClientRect().width ) }px 0px`,
 		} );
 
 		observer.observe( picker );
 
-		// Add the colors swatches.
-		Object.entries( this.colors ).map( ( [ control, value ] ) => {
-			const swatch = document.createElement( 'div' );
-			swatch.classList.add( `${ this.pickerClass }__swatch` );
-			swatch.style = `--color: ${ value }`;
-			swatch.dataset.text = value.replace( '#', '' );
-
-			swatch.addEventListener( 'mouseenter', () => {
-				$e.run( 'elements-color-picker/enter-preview', { value } );
-			} );
-
-			swatch.addEventListener( 'mouseleave', () => {
-				$e.run( 'elements-color-picker/exit-preview' );
-			} );
-
-			swatch.addEventListener( 'click', ( e ) => {
-				$e.run( 'elements-color-picker/apply', {
-					value,
-					trigger: {
-						palette: picker,
-						swatch: e.target,
-					},
-				} );
-
-				e.stopPropagation();
-			} );
-
-			picker.append( swatch );
-		} );
+		if ( 0 === count ) {
+			// Show a Tipsy tooltip.
+			this.addTooltip( picker );
+		} else {
+			// Add the colors swatches.
+			this.addColorSwatches( picker );
+		}
 
 		// Remove the picker on mouse leave.
-		this.container.view.$el[ 0 ].addEventListener( 'mouseleave', function handler( e ) {
-			e.currentTarget.removeEventListener( 'mouseleave', handler );
+		this.container.view.$el[ 0 ].addEventListener( 'mouseleave', () => {
+			jQuery( picker ).tipsy( 'hide' );
```

Comment:
Can it be extracted to method `removeTooltip` to be consistent with the method `addTooltip` + it will be easier to refactor it later and use some other library for the tooltip mechanism.

Reply:
Yup, good idea :)